eZ/Publish/Core/MVC/Symfony/Controller/Content/QueryController.php

```diff
@@ -81,4 +87,54 @@ private function runQuery(ContentView $view, $method)
         );
         $view->addParameters([$view->getParameter('query')['assign_results_to'] => $searchResults]);
     }
+
+    /**
+     * @param ContentView $view
+     * @param Request $request
+     * @return ContentView
+     */
+    public function pagingQueryAction(ContentView $view, Request $request)
+    {
+        $this->runPagingQuery($view, $request);
+
+        return $view;
+    }
+
+    /**
+     * @param ContentView $view
+     * @param Request $request
+     */
+    private function runPagingQuery(ContentView $view, Request $request)
+    {
+        $pager = new Pagerfanta(
+            $this->getAdapter($this->contentViewQueryTypeMapper->map($view))
+        );
+
+        $pager->setMaxPerPage($view->getParameter('query')['limit']);
+
+        $requestPageName = 'page';
+        if (array_key_exists('request_page_name', $view->getParameter('query'))) {
```

Comment:
@alongosz Suggestions on name here?
Was thinking `request_page_param` or `page_param` or something else making it clear what it is.

Reply:
page_param will do.